Automattic/jetpack
# Update VS Code configuration

Currently, with our VS Code configurations, the lints reported by ESLint and PHP Code Sniffer are not fixed automatically on save, rather we rely on pre-commit hooks to do it for us.

## Proposed changes:

* Enable auto-formatting on save for PHP and ES files
* Add the lining and formatting extensions to the recommended extensions list

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* On `trunk`, remove all your user settings in VS Code
* Add some lints in PHP and JS/TS files - e.g. mess up spaces
* Save the file
* Lints are not fixed
* Now checkout this PR
* Goto VS Code extensions section and type `@recommended` in the search input
* Ensure that you have all the recommended extensions installed
* Reload the window <kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd> > "Developer: Reload Window"
* Now add some lints as mentioned above and save
* Confirm that the lints are fixed
